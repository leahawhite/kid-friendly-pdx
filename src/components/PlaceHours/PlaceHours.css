.place-hours {
  display: flex;
  justify-content: left;
  margin: 20px 10px 20px 0;
  padding-bottom: 20px;
  border-bottom: 1px solid gray;
}

.place-hours-icon {
  display: flex;
  width: 10%;
  justify-content: center;
  align-self: center;
  margin-left: 5%;
  margin-right: 5%;
  padding: 0;
} 

.place-hours-content {
  text-align: left;
  width: 89%;
}

.place-hours-table {
  border-collapse: collapse;
  border-spacing: 0;
}

.place-hours-table-column-left {
  min-width: 110px;
  padding-right: 10px;
}
