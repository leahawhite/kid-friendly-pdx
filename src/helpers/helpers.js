/*export function numberOfReviews(place) {
  const placeReviews = reviews.filter(review.place_id == place.id)
  return placereviews.length
}

export function averageReviewRating(place) {
  const placeReviews = reviews.filter(review.place_id == place.id)
  const ratingArray = placeReviews.map(review => review.rating) 
  const ratingsSum = ratingArray.reduce(function(a,b){return a + b})
  const numReviews = this.numberOfReviews(place)
  return ratingsSum / numReviews
}*/

export function readableReviewCount(number) {
  switch(number) {
    case 0:
      return 'no reviews yet'

    case 1:
      return `1 review`

    default:
      return `${number} reviews`
  }
}

export const testPlace = {
  id: 1,
  name: "Atlas Pizza",
  address: "6529 SE Foster Rd",
  city: "Portland",
  state: "OR",
  zipcode: "97206",
  lat: 45.489200,
  lng: -122.595070,
  neighborhood: "SE",
  phone: "503-232-3004",
  website: "http://atlaspizzapdx.com",
  hours: [
    {
      "dayOfWeek": "Monday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Tuesday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Wednesday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Thursday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Friday",
      "opens": "12:00 pm",
      "closes": "12:00 am",
    },
    {
      "dayOfWeek": "Saturday",
      "opens": "12:00 pm",
      "closes": "12:00 am",
    },
    {
      "dayOfWeek": "Sunday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
  ],
  date_created: "2019-06-13",
  category: ["restaurants",],
  descriptors: [ "pizza", "arcade", "beer" ],
  features: { 
    "toys": true, 
    "play area": true, 
    "arcade": true,
    "quick service": true,
    "all ages": true,
    "comfortable seating": true,
    "friendly staff": true,
    "kids menu": false,
    "highchairs/boosters": false,
    "changing station": false,
    "flexible": true,
    "patio/sidewalk": true,
  },
  images: [
    {
      id: 1,
      src: 'https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400',
      title: `Atlas Pizza's 'Jalapeno Popper' pie`,
      place_id: 1,
      user_id: 2,
      review_id: 1,
    },
    {
      id: 2,
      src: 'https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400',
      title: 'A rainbow over Foster Blvd',
      place_id: 1,
      user_id: 1,
      review_id: 2,
    },
  ],
  number_of_reviews: 2,
  average_review_rating: 4.5,
};

export const testReviews = [
  { 
    id: 1,
    star_rating: 5,
    text: "Atlas is our new favorite spot for a quick, painless dinner with our toddler. It's super casual -- order slices or pies at the counter, and they call your name for pickup when it's ready. Since it's new, the play area is pretty clean and well-stocked. There's an old church pew there for parents to sit and supervise, but the open floor plan means that you can see the play area from pretty much every table. We've only ordered slices, so we've never had to wait long. We're usually in and out in 30 minutes, which is just about perfect for us.",
    date_created: "2019-06-13",
    place_id: 1,
    user_id: 5,
    images: [
      {
        id: 1,
        src: 'https://lh3.googleusercontent.com/QDwk6G6ZCwYzVH-yJHQ824GZ1wKW2XXus8hxxdH-4AAHz-FejC-UVydyE5le1ZvPTIo1cXw1G25qhljwS8dgyIedCCFtIxBonznVQ4LfaPDEZVWmqKS1vFqVUFFNmMmMlKQ5eCpfhNI=w2400',
        title: `Atlas Pizza's play area`,
        place_id: 1,
        user_id: 5,
      },
    ]
  },
  { 
    id: 2,
    star_rating: 4,
    text: "Very easy place to take kids of all ages. Nice toy area with a great variety of toys!",
    date_created: "2019-06-13",
    place_id: 1,
    user_id: 1,
    images: [],
  },
];



