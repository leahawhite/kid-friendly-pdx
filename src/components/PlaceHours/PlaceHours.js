import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import moment from 'moment'
import './PlaceHours.css'

export default class PlaceHours extends Component {
  static defaultProps = {
    place: {}
  }

  getStdTime(hours) {
    return moment(hours, 'hh:mm a').format('hh:mm a')
  }

  render() {
    const { place } = this.props || {}
    // converting dayOfWeek value to text equivalent
    const newPlaceHours = place && place.hours && place.hours.length && place.hours.map(day => { 
      let returnValue = {...day}
      if (day.dayOfWeek === 1) {
        returnValue.dayOfWeek = "Monday";
      }
      if (day.dayOfWeek === 2) {
        returnValue.dayOfWeek = "Tuesday";
      }
      if (day.dayOfWeek === 3) {
        returnValue.dayOfWeek = "Wednesday";
      }
      if (day.dayOfWeek === 4) {
        returnValue.dayOfWeek = "Thursday";
      }
      if (day.dayOfWeek === 5) {
        returnValue.dayOfWeek = "Friday";
      }
      if (day.dayOfWeek === 6) {
        returnValue.dayOfWeek = "Saturday";
      }
      if (day.dayOfWeek === 7) {
        returnValue.dayOfWeek = "Sunday";
      }
      return returnValue
    })
   
    const hoursArr = newPlaceHours && newPlaceHours.length && newPlaceHours.map((item, index) =>
      <table className="place-hours-table" key={index}>
        <tbody>
          <tr className="place-hours-table-row">
            <th className="place-hours-table-column-left">{item.dayOfWeek}{': '}</th>
            <td className="place-hours-table-column-right">{this.getStdTime(item.opens)}{' - '}{this.getStdTime(item.closes)}</td>
          </tr>
        </tbody>
      </table>  
    );
    return (
      <div className="place-hours">
        <div className="place-hours-icon">
          <FontAwesomeIcon icon="clock" size="lg" />
        </div>
        <div className="place-hours-content">
          {hoursArr}
        </div>
      </div>
    )
  }
}