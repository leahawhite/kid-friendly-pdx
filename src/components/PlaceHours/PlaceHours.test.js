import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PlaceHours from './PlaceHours';
import { testPlace } from '../../helpers/helpers';

describe('PlaceHours component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PlaceHours />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected with props', () => {
    const wrapper = shallow(<PlaceHours place={testPlace} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})