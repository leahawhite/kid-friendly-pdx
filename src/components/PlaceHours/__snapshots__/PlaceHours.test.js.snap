// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceHours component renders UI as expected with props 1`] = `
<div
  className="place-hours"
>
  <div
    className="place-hours-icon"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="clock"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="lg"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="place-hours-content"
  >
    <table
      className="place-hours-table"
      key="0"
    >
      <tbody>
        <tr
          className="place-hours-table-row"
        >
          <th
            className="place-hours-table-column-left"
          >
            Monday
            : 
          </th>
          <td
            className="place-hours-table-column-right"
          >
            12:00 pm
             - 
            11:00 pm
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="place-hours-table"
      key="1"
    >
      <tbody>
        <tr
          className="place-hours-table-row"
        >
          <th
            className="place-hours-table-column-left"
          >
            Tuesday
            : 
          </th>
          <td
            className="place-hours-table-column-right"
          >
            12:00 pm
             - 
            11:00 pm
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="place-hours-table"
      key="2"
    >
      <tbody>
        <tr
          className="place-hours-table-row"
        >
          <th
            className="place-hours-table-column-left"
          >
            Wednesday
            : 
          </th>
          <td
            className="place-hours-table-column-right"
          >
            12:00 pm
             - 
            11:00 pm
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="place-hours-table"
      key="3"
    >
      <tbody>
        <tr
          className="place-hours-table-row"
        >
          <th
            className="place-hours-table-column-left"
          >
            Thursday
            : 
          </th>
          <td
            className="place-hours-table-column-right"
          >
            12:00 pm
             - 
            11:00 pm
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="place-hours-table"
      key="4"
    >
      <tbody>
        <tr
          className="place-hours-table-row"
        >
          <th
            className="place-hours-table-column-left"
          >
            Friday
            : 
          </th>
          <td
            className="place-hours-table-column-right"
          >
            12:00 pm
             - 
            12:00 am
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="place-hours-table"
      key="5"
    >
      <tbody>
        <tr
          className="place-hours-table-row"
        >
          <th
            className="place-hours-table-column-left"
          >
            Saturday
            : 
          </th>
          <td
            className="place-hours-table-column-right"
          >
            12:00 pm
             - 
            12:00 am
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="place-hours-table"
      key="6"
    >
      <tbody>
        <tr
          className="place-hours-table-row"
        >
          <th
            className="place-hours-table-column-left"
          >
            Sunday
            : 
          </th>
          <td
            className="place-hours-table-column-right"
          >
            12:00 pm
             - 
            11:00 pm
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
